core/src/crypto/DeterministicPublicKey.cpp

```diff
@@ -113,9 +113,8 @@ namespace ledger {
                 throw Exception(api::ErrorCode::UNSUPPORTED_OPERATION, "Cannot derive key - IL >= N");
             }
 
-            SECP256k1Point K = SECP256k1Point(_key).generatorMultiply(IL.toByteArray());
             return DeterministicPublicKey(
-                    K.toByteArray(),
+                    SECP256k1Point(_key).generatorMultiply(IL.toByteArray()).toByteArray(),
```

Comment:
I don’t get why that change.

Reply:
To not have `SECP256k1Point K = blablabla`, and avoid having copies (now everything are moving), but actually now when I add move constructor it should be fine.

Reply:
You got a great point here but when I read that portion of code, the move mechanism isn't obvious.
I suggest you to keep the `K` variable and add an appropriate `std::move` in the constructor.

Reply:
Move it back